examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/edit/bloc/puppy_edit_form_event.dart

```diff
@@ -0,0 +1,13 @@
+part of 'puppy_edit_form_bloc.dart';
+
+@immutable
+abstract class PuppyEditFormEvent extends FormBlocEvent{}
+
+class PuppyEditFormSetImageEvent extends PuppyEditFormEvent {
+  PuppyEditFormSetImageEvent({required this.source});
+
+  final ImagePickerAction source;
```

Comment:
@StanevPrime I thought to use a custom event containing ImagePickerAction and a custom State PuppyEditFormState 
 containing a String with the new image asset path. I only found this issue https://github.com/GiancarloCode/form_bloc/issues/142 about adding events and states to the form_bloc.